src/DGtal/shapes/parametric/Astroid2D.ih

```diff
@@ -84,27 +84,29 @@ DGtal::Astroid2D<T>::parameter( const RealPoint2D & pp ) const
   p -= myCenter;
   double angle;
 
-  if ( p[0] == 0. && p[1] > 0. )
-    angle = M_PI/2.0;
-  else if ( p[0] == 0. && p[1] < 0. )
-    angle = 3.0*M_PI/2.0;
-  else if ( p[1] == 0. && p[0] >= 0. )
-    angle = 0.0;
-  else if ( p[1] == 0. && p[0] < 0. )
-    angle = M_PI;
+  if ( p[0] == 0. ) {
```

Comment:
Done by adding a local method `isAlmostEqual` at the top of the file.
Maybe this could be considered as a more global helper in DGtal?

Reply:
@akrah It seems to me that something like this will be good to have globally. Ping @dcoeurjo

Reply:
having a global `isAlmostEqual` seems very dangerous to me. It is in catch for testing purposes. for any other usages (like this one), it is better that the author controls its code arithmetics compared to rely on a global function (that would need a magic number at some point)

Reply:
@dcoeurjo I think that the danger is not to propose this helper method that will have probably the same implementation in most of cases. Indeed, this could save time for users and limit implementation errors.

Moreover, even with this method proposed by DGtal, an author who wants use its own method can do it. The most important for me is to warn users about the implementation details in the documentation.

Such method could be inspired from these links :
- https://en.cppreference.com/w/cpp/types/numeric_limits/epsilon
- https://stackoverflow.com/a/253874

Reply:
For me, checking if two doubles are equal is a design problem ;) and when needed, it should be carefully handled by the author with a specific solution.

Again it makes sense here or in unit test, but not in general. The `std::numeric_limits<T>::epsilon()` cannot be a solution because with doubles, the distance between two representable numbers is not the same close to zero or for very huge numbers. 

I do prefer to handle it at PR level when reviewing the code.;)